tools/edid/generate-rpi-edid-dumps

```diff
@@ -70,107 +101,363 @@ def run_cmd(cmd):
+    for model in screen_models_with_md5:
+        raw_edid_path = raw_edid_path_pattern.format(filename=model)
+        edid_dumps_path = edid_dumps_path_pattern.format(filename=model)
+        expected_data_path = expected_data_path_pattern.format(filename=model)
+
+        # Avoid writing over existing data without user intention.
+        if not override_mode:
+            if os.path.exists(edid_dumps_path) or os.path.exists(expected_data_path):
+                verbose_print(
+                    'Not overriding {} and {}, skipping EDID.'
+                    .format(edid_dumps_path, expected_data_path)
+                )
+                continue
+
+        verbose_print('Working on {}'.format(raw_edid_path))
+
+        # Flash the screen with the current EDID.
+        if not flash_screen(raw_edid_path):
+            return RC_ERR_SUBPROCESS
+
+        flashed_screen = True
+
+        # Generate the RPi and kano-settings specific data.
+        edid_dumps = generate_rpi_dumps(raw_edid_path, edid_dumps_path)
+        if not edid_dumps:
+            return RC_ERR_SUBPROCESS
+
+        if not generate_expected_data(edid_dumps, expected_data_path):
+            return RC_ERR_SUBPROCESS
+
+    if flashed_screen:
+        if not restore_original_edid():
+            return RC_ERR_SUBPROCESS
+
+    return RC_SUCCESS
+
+
+def check_python_modules(modules):
+    """ Check that the given Python modules are installed. """
+
+    has_all_modules = True
+
+    for module in modules:
+        try:
+            importlib.import_module(module)
+        except ImportError:
+            sys.stderr.write('Error: Missing dependency on Python module {}\n'.format(module))
+            has_all_modules = False
+
+    return has_all_modules
+
+
+def check_installed_tools(binary_paths):
+    """ Check that the given binaries are installed. """
+
+    has_all_binaries = True
+
+    for binary_path in binary_paths:
+        if not os.path.exists(binary_path):
+            sys.stderr.write('Error: Missing dependency on binary {}\n'.format(binary_path))
+            has_all_binaries = False
+
+    return has_all_binaries
+
+
+def check_boot_config_txt():
+    """ Check that the /boot/config.txt is setup correctly for this tool to work. """
+
+    valid_config_txt = True
+
+    with open('/boot/config.txt', 'r') as config_txt:
+        config_txt_lines = config_txt.readlines()
+
+    if 'hdmi_force_hotplug=1\n' in config_txt_lines:
+        sys.stderr.write(
+            'Error: hdmi_force_hotplug=1 was found in /boot/config.txt\n'
+        )
+        valid_config_txt = False
+
+    if 'dtparam=i2c2_iknowwhatimdoing=on\n' not in config_txt_lines:
+        sys.stderr.write(
+            'Error: dtparam=i2c2_iknowwhatimdoing=on was not found in /boot/config.txt\n'
+        )
+        valid_config_txt = False
+
+    return valid_config_txt
+
+
+def check_i2c_bus(bus_number):
+    """ Check that the i2c device at address 3f805000.i2c is available. """
+
```

Comment:
This address is provided by `i2c_bcm2708` kernel module right?
Can we mention it here? Does it need a check using a `lsmod` or similar?

Reply:
This one is required for the `edid-rw` tool to work properly on RPi. The i2c address to write to the HDMI port does not change. 

There are comments and a hint to use `i2cdetect -l` to check that it's available. What other checks would be necessary? (not sure I understand your questions)

Reply:
Oh I understand now. I wasn't sure about the purpose of the address that's all.